src/Appwrite/Platform/Tasks/ScheduleBase.php

```diff
@@ -127,76 +128,74 @@ public function action(Group $pools, Database $dbForConsole, callable $getProjec
-                        $new = \strtotime($document['resourceUpdatedAt']);
-
-                        if (!$document['active']) {
-                            Console::info("Removing: {$document['resourceId']}");
-                            unset($this->schedules[$document->getInternalId()]);
-                        } elseif ($new !== $org) {
-                            Console::info("Updating: {$document['resourceId']}");
-                            $this->schedules[$document->getInternalId()] = $getSchedule($document);
-                        }
-                    }
+            while ($sum === $limit) {
+                $paginationQueries = [Query::limit($limit)];
 
-                    $latestDocument = \end($results);
+                if ($latestDocument) {
+                    $paginationQueries[] = Query::cursorAfter($latestDocument);
                 }
 
-                $lastSyncUpdate = $time;
-                $timerEnd = \microtime(true);
+                $results = $dbForConsole->find('schedules', \array_merge($paginationQueries, [
+                    Query::equal('region', [System::getEnv('_APP_REGION', 'default')]),
+                    Query::equal('resourceType', [static::getSupportedResource()]),
+                    Query::greaterThanEqual('resourceUpdatedAt', $lastSyncUpdate),
+                ]));
 
-                $pools->reclaim();
+                $sum = count($results);
+                $total = $total + $sum;
 
-                Console::log("Sync tick: {$total} schedules were updated in " . ($timerEnd - $timerStart) . " seconds");
-            });
+                foreach ($results as $document) {
+                    $localDocument = $schedules[$document['resourceId']] ?? null;
 
-            Timer::tick(
-                static::ENQUEUE_TIMER * 1000,
-                fn () => $this->enqueueResources($pools, $dbForConsole)
-            );
+                    // Check if resource has been updated since last sync
+                    $org = $localDocument !== null ? \strtotime($localDocument['resourceUpdatedAt']) : null;
+                    $new = \strtotime($document['resourceUpdatedAt']);
 
-            $this->enqueueResources($pools, $dbForConsole);
+                    if (!$document['active']) {
+                        Console::info("Removing: {$document['resourceId']}");
+                        unset($this->schedules[$document->getInternalId()]);
+                    } elseif ($new !== $org) {
+                        Console::info("Updating: {$document['resourceId']}");
+                        $this->schedules[$document->getInternalId()] = $getSchedule($document);
+                    }
+                }
+
+                $latestDocument = \end($results);
+            }
+
+            $lastSyncUpdate = $time;
+            $timerEnd = \microtime(true);
+
+            $connections->reclaim();
```

Comment:
where is connection defined ?

Reply:
Here https://github.com/appwrite/appwrite/blob/1581a8c654e17ab1f44673866f30c62a76157e8e/src/Appwrite/Platform/Tasks/ScheduleBase.php#L139

This is a local connections variable